packages/storybook-builder-vite/codegen-importfn-script.js

```diff
@@ -0,0 +1,48 @@
+const { normalizeStories } = require('@storybook/core-common');
+
+function toImportFnPart(specifier, storyImports) {
+    const { directory, importPathMatcher } = specifier;
+
+    return `
+      async (path) => {
+        if (!${importPathMatcher}.exec(path)) {
+          return;
+        }
+        
+        const pathRemainder = path.substring(${directory.length + 1});
+        const module = (await import('${storyImports}')).default;
+        return module[path]();
```

Comment:
@joshwooding I think you could cut out the middle man here and just put your hash from `codegen-virtual-imports-script.js` in here directly:

```js
const importers = {
  'x': async() => import('x'), // one per CSF file, as you do below
}

const toImportFn = async (path) => importers[path](); 
```

We might eventually do something similar in webpack too, in which case we should definitely try and share code for this generation,.

The complex code we have for the webpack (`importFnPart` etc) is to trigger webpack's glob matching behaviour which you don't need AFIACT (?)

Reply:
Great idea. This all seems to work to me. I was trying to stick closely to how webpack does it to avoid confusion but it's much cleaner this way.